### ISSUE TITLE
Plugin: Parsing string array in nested config structs doesn't work

### ISSUE THREAD

**Issue by sevensolutions** [traefik/traefik#10996]
### Welcome!

- [X] Yes, I've searched similar issues on [GitHub](https://github.com/traefik/traefik/issues) and didn't find any.
- [X] Yes, I've searched similar issues on the [Traefik community forum](https://community.traefik.io) and didn't find any.

### What did you do?

I'am working on a traefik plugin where i want to configure the plugin using a yaml-array.
This works fine on top-level, but not on a nested one.

I found the following depending PRs:
- https://github.com/traefik/traefik/pull/8886
- https://github.com/traefik/paerser/pull/15

They all talk about traefik 2.x but i'am using 3.1.
Is there something missing in v3?

Here is my code:
**Config:**
```yml
middlewares:
    oidc-auth:
      plugin:
        traefik-oidc-auth:
          Authorization:
            AssertClaims:
              - Name: "preferred_username"
                Values: ["value1", "value2", "value3"]
```

**Structs:**
```go
type Config struct {
	Authorization *AuthorizationConfig `json:"authorization"`
}
type AuthorizationConfig struct {
	AssertClaims []ClaimAssertion `json:"assert_claims"`
}
type ClaimAssertion struct {
	Name   string   `json:"name"`
	Value  string   `json:"value"`
	Values []string `json:"values"`
}
```

### What did you see instead?

When i loop over the Values-array, i get: `[║24║value1║value2║value3]`
Instead, i'am expecting to get a real array.
The only workaround is to use:
```yml
Values: "value1,value2,value3"
```

### What version of Traefik are you using?

Version:      3.1.2
Codename:     comte
Go version:   go1.22.5
Built:        2024-08-06T13:37:51Z
OS/Arch:      linux/amd64

### What is your environment & configuration?

```yaml
global:
  checkNewVersion: true
  sendAnonymousUsage: true

entryPoints:
  web:
    address: :80

# Enable API and dashboard
api:
  insecure: true
  dashboard: true
  debug: true

experimental:
  localPlugins:
    traefik-oidc-auth:
      moduleName: github.com/sevensolutions/traefik-oidc-auth

providers:
  file:
    filename: /etc/traefik/configs/http.yml
    watch: true
```

**Comment by nmengin** [comment]
Hello @sevensolutions,

Thanks for the bug report, we were able to reproduce it. We would love community support to address it.  

If you or another community member would like to build it, let us know, and we will work with you to make sure you have all the information needed so that it can be merged.  

We prefer to work with our community members at the beginning of the design process so that we can make sure that we are aligned and can move quickly with the review and merge process.  Let us know here or create a PR before you start, and we will work with you there.  

Don’t forget to check out the (contributor docs)[https://doc.traefik.io/traefik/contributing/submitting-pull-requests/] and to link the PR to this issue.

**Comment by sevensolutions** [comment]
Hi @nmengin,
i've cloned the repo and started analyzing this a bit but i don't really understand it.

It looks like this exact problem has already been fixed in the paerser-library in [this commit](https://github.com/traefik/paerser/commit/368ce2e07a1d7b673fe92bec9e5b752c8a73e710)

I've then checked which version of paerser is used by traefik, which is 0.2.0. This already includes the fix.

It looks like traefik 3 has the same problem which has been fixed in #8886 so i think we just need to apply the same changes to version 3. Maybe this should have been merged but haven't?

I can try to implement these changes and make some tests...
